src/Symfony/Bundle/SecurityBundle/Resources/config/security_debug.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Bundle\SecurityBundle\Debug\TraceableFirewallListener;
+use Symfony\Bundle\SecurityBundle\EventListener\VoteListener;
+use Symfony\Component\Security\Core\Authorization\TraceableAccessDecisionManager;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('debug.security.access.decision_manager', TraceableAccessDecisionManager::class)
+            ->decorate('security.access.decision_manager')
+            ->args([
+                service('debug.security.access.decision_manager.inner'),
+            ])
+
+        ->set('debug.security.voter.vote_listener', VoteListener::class)
+            ->tag('kernel.event_subscribe')
+            ->args([
+                service('debug.security.access.decision_manager'),
+            ])
+
+        ->set('debug.security.firewall', TraceableFirewallListener::class)
+            ->tag('kernel.event_subscribe')
+            ->args([
+                service('security.firewall.map'),
+                service('event_dispatcher'),
+                service('security.logout_url_generator'),
+            ])
+        ->alias('debug.security.firewall', 'security.firewall')
```

Comment:
This alias is maybe badly written.
```xml
<service id="security.firewall" alias="debug.security.firewall" />
```

it's maybe 
```php
 ->alias('security.firewall', 'debug.security.firewall')
```
 but tests failed any idea ?